samples/benchmarks/coremark/README.rst

```diff
@@ -0,0 +1,155 @@
+.. _coremark_sample:
+
+CoreMark
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The sample demonstrates how to run the `CoreMark®`_ benchmark to evaluate performance of a core.
+
+.. _CoreMark®: https://www.eembc.org/coremark/
+
+Requirements
+************
+
+The sample supports the following development kits:
+
++--------------------------------+-----------+------------------------------------------------+------------------------------------+
+|Hardware platforms              |PCA        |Board name                                      |Build target                        |
++================================+===========+================================================+====================================+
+|`nRF52 DK`_                     |PCA10040   |:ref:`nrf52dk_nrf52832 <nrf52dk_nrf52832>`      |``nrf52dk_nrf52832``                |
++--------------------------------+-----------+------------------------------------------------+------------------------------------+
+|`nRF52840 DK`_                  |PCA10056   |:ref:`nrf52840dk_nrf52840 <nrf52840dk_nrf52840>`|``nrf52840dk_nrf52840``             |
++--------------------------------+-----------+------------------------------------------------+------------------------------------+
+|`nRF5340 DK`_                   |PCA10095   |:ref:`nrf5340dk_nrf5340 <nrf5340dk_nrf5340>`    |``nrf5340dk_nrf5340_cpuapp``        |
++--------------------------------+-----------+------------------------------------------------+------------------------------------+
+Overview
+********
+
+The sample runs the CoreMark benchmark on the target CPU.
+CoreMark evaluates the CPU efficiency by performing different algorithms, such as state machine, CRC calculation, matrix manipulation and list processing (find and sort).
+
+To run CoreMark benchmark on the preferred core, press the button that corresponds to the core.
+When the benchmark finishes running, it can be started again by pressing the button assigned to the given core.
+Enable ``COREMARK_MODE_FLASH_AND_RUN`` if you want the sample to run for the first time upon startup.
+
+User interface
+**************
+
+Each target CPU has an assigned button and LED that are responsible for triggering the benchmark's start and ``test in progress``:
+
+Button 1:
+   Start the benchmark run on the Application Core.
+Button 2:
+   Start the benchmark run on the Network/Radio Core.
+Button 3:
+   Start the benchmark run on the PPR Core.
```

Comment:
Right now there is no nRF54H related code in this PR so maybe we should also remove such mentions from docs? We could introduce it later with proper support